cli: TestHalfOnlineLossOfQuorumRecovery failed

cli.TestHalfOnlineLossOfQuorumRecovery [failed](https://tanzanite.cluster.engflow.com/invocations/default/19c59fd8-74b2-488c-98d1-c5bf93a4d383?testReportRun=17&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
=== RUN   TestHalfOnlineLossOfQuorumRecovery
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestHalfOnlineLossOfQuorumRecovery1785807376
    test_log_scope.go:76: use -show-logs to present logs inline
[debug recover make-plan --confirm=y --certs-dir=test_certs --host=127.0.0.1:38653 --plan=/var/lib/engflow/worker/work/0/exec/_tmp/9273343b73f47135f2171c51f2f417b7/TestHalfOnlineLossOfQuorumRecovery2597561690/recovery-plan.json]
[debug recover apply-plan --certs-dir=test_certs --host=127.0.0.1:38653 --confirm=y /var/lib/engflow/worker/work/0/exec/_tmp/9273343b73f47135f2171c51f2f417b7/TestHalfOnlineLossOfQuorumRecovery2597561690/recovery-plan.json]
[debug recover verify --certs-dir=test_certs --host=127.0.0.1:38653 /var/lib/engflow/worker/work/0/exec/_tmp/9273343b73f47135f2171c51f2f417b7/TestHalfOnlineLossOfQuorumRecovery2597561690/recovery-plan.json]
[debug recover verify --certs-dir=test_certs --host=127.0.0.1:38653 /var/lib/engflow/worker/work/0/exec/_tmp/9273343b73f47135f2171c51f2f417b7/TestHalfOnlineLossOfQuorumRecovery2597561690/recovery-plan.json]
    debug_recover_loss_of_quorum_test.go:643: 
        	Error Trace:	
        	Error:      	"debug recover verify --certs-dir=test_certs --host=127.0.0.1:38653 /var/lib/engflow/worker/work/0/exec/_tmp/9273343b73f47135f2171c51f2f417b7/TestHalfOnlineLossOfQuorumRecovery2597561690/recovery-plan.json\nChecking application of recovery plan 9dc3772f-52d3-4882-8d1e-f3719e8e0fa5\nUnavailable ranges:\n r1 : loss of quorum, key span /{Min-System/NodeLiveness}\n r2 : loss of quorum, key span /System/NodeLiveness{-Max}\n r3 : loss of quorum, key span /System/{NodeLivenessMax-tsd}\n r4 : loss of quorum, key span /System{/tsd-tse}\n r5 : loss of quorum, key span /{Systemtse-Table/0}\n r6 : loss of quorum, key span /Table/{0-3}\n r7 : loss of quorum, key span /Table/{3-4}\n r8 : loss of quorum, key span /Table/{4-5}\n r9 : loss of quorum, key span /Table/{5-6}\n r10 : loss of quorum, key span /Table/{6-7}\n r11 : loss of quorum, key span /Table/{7-8}\n r12 : loss of quorum, key span /Table/{8-9}\n r13 : loss of quorum, key span /Table/{9-11}\n r14 : loss of quorum, key span /Table/1{1-2}\n r15 : loss of quorum, key span /Table/1{2-3}\n r16 : loss of quorum, key span /Table/1{3-4}\n r17 : loss of quorum, key span /Table/1{4-5}\n r18 : loss of quorum, key span /Table/1{5-6}\n r19 : loss of quorum, key span /Table/1{6-7}\n r20 : loss of quorum, key span /Table/1{7-8}\nFailed to complete range health check: found more failed ranges than limit 20\nRecovery plan application progress:\n plan applied successfully on node n1\nAll dead nodes are decommissioned.\nERROR: loss of quorum recovery did not fully succeed\n" does not contain "Loss of quorum recovery is complete."
        	Test:       	TestHalfOnlineLossOfQuorumRecovery
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestHalfOnlineLossOfQuorumRecovery1785807376
--- FAIL: TestHalfOnlineLossOfQuorumRecovery (74.88s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=17</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHalfOnlineLossOfQuorumRecovery.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45157

--- a/pkg/cli/debug_recover_loss_of_quorum_test.go
+++ b/pkg/cli/debug_recover_loss_of_quorum_test.go
@@ -462,6 +462,10 @@ func TestHalfOnlineLossOfQuorumRecovery(t *testing.T) {
 	listenerReg := listenerutil.NewListenerRegistry()
 	defer listenerReg.Close()
 
+	st := cluster.MakeTestingClusterSettings()
+	// We currently don't clear out the LeadEpoch field when recovering from a
+	// loss of quorum, so we can't run with leader leases on in this test.
+	kvserver.OverrideLeaderLeaseMetamorphism(ctx, &st.SV)
 	// Test cluster contains 3 nodes that we would turn into a single node
 	// cluster using loss of quorum recovery. To do that, we will terminate
 	// two nodes and run recovery on remaining one. Restarting node should
@@ -476,6 +480,7 @@ func TestHalfOnlineLossOfQuorumRecovery(t *testing.T) {
 	sa := make(map[int]base.TestServerArgs)
 	for i := 0; i < 3; i++ {
 		sa[i] = base.TestServerArgs{
+			Settings: st,
 			Knobs: base.TestingKnobs{
 				Server: &server.TestingKnobs{
 					StickyVFSRegistry: fs.NewStickyRegistry(),

